### PR TITLE
verify ErrCodeStoreOutOfGas fix in e2etest

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: blockchain/integrity/integrity_test.go [hunk 1]
```diff
@@ -763,29 +841,59 @@ func TestConstantinople(t *testing.T) {
...
+				bLog, err := dao.TransactionLogs(blkHeight)
+				require.NoError(err)
+				tLog := bLog.Logs[0]
+				// first transaction log is gas fee
+				tx := tLog.Transactions[0]
+				require.Equal(tx.Sender, caller.String())
+				require.Equal(tx.Recipient, address.RewardingPoolAddr)
+				require.Equal(iotextypes.TransactionLogType_GAS_FEE, tx.Type)
+
+				hu := config.NewHeightUpgrade(&cfg.Genesis)
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:868]: 3 tx log pre-Greenland

File: blockchain/integrity/integrity_test.go [hunk 2]
```diff
@@ -763,29 +841,59 @@ func TestConstantinople(t *testing.T) {
...
+				if hu.IsPre(config.Greenland, blkHeight) {
+					// pre-Greenland contains a tx with status = ReceiptStatus_ErrCodeStoreOutOfGas
+					// due to a bug the transfer is not reverted
+					require.Equal(3, len(tLog.Transactions))
+					// 2nd and 3rd are in-contract-transfer
+					tx = tLog.Transactions[1]
```
Comment (dustinxie) [inline blockchain/integrity/integrity_test.go:874]: 3 tx log pre-Greenland, and 1 tx log post-Greenland

File: blockchain/integrity/integrity_test.go [hunk 3]
```diff
@@ -159,18 +168,53 @@ func addTestingConstantinopleBlocks(bc blockchain.Blockchain, dao blockdao.Block
...
...657446726f7a656e506572696f64206265666f7265207363686564756c652074616b65206f66662074696d65636f6e747261637420737461747573206e65656420746f2062652077616974696e6720666f72206469737075746573686f756c64206368616e676520616674657220287363686564756c655f74616b655f6f66665f74696d65202b20656e645f7265706f72745f696e74657276616c296f6e6c7920726567697374726564206f7261636c652063616e207265706f727420726573756c742173686f756c6420736574746c6520616674657220287363686564756c655f74616b655f6f66665f74696d65202b20656e645f646973707574655f696e74657276616c29a265627a7a72315820612ed13838b4a948337cce571ea68c9fc0e970d7cf3fb2271f2606a6dc6b4f4264736f6c634300050b00327265616c206d61782062656e656669742073686f756c6420657175616c20746f206d61782062656e6566697421696e73756666696369656e7420696e697469616c206465706f7369742066756e64217363686564756c652074616b65206f66662074696d652073686f756c64206174206c65617374206d61782062656e656669742073686f756c64206c6172676572207468616e207072656d69756d00000000000000000000000000000000000000000000021e19e0c9bab2400000323031392d31302d313700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005da9004400000000000000000000000079f3574d440c21029eecb8e9ee66b23b26cbcb38")
```
Comment (yutongp) [inline blockchain/integrity/integrity_test.go:186]: put it in a separate file

Comment (dustinxie) [reply blockchain/integrity/integrity_test.go]: fixed

**Pull request merged**